File: src/1207.unique-number-of-occurrences.1319/solution.ts
function uniqueOccurrences (arr: number[]): boolean {
    const numCountMap = new Map<number, number>();
    for (let i = 0; i < arr.length; i++) {
        if (numCountMap.has(arr[i])) {
            numCountMap.set(arr[i], <number>numCountMap.get(arr[i]) + 1);
        } else {
            numCountMap.set(arr[i], 1);
        }
    }
    const set = new Set<number>();
    for (const count of numCountMap.values()) {
        if (set.has(count)) {
            return false;
        }
        set.add(count);
    }
    return true;
}

File: src/1207.unique-number-of-occurrences.1319/solution.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        int[] counts = new int[2001];
        for (int k : arr) {
            counts[k + 1000]++;
        }
        Set<Integer> set = new HashSet<>();
        for (int count : counts) {
            if (count == 0) {
                continue;
            }
            if (set.contains(count)) {
                return false;
            }
            set.add(count);
        }
        return true;
    }
}